java-frontend/src/main/java/org/sonar/java/se/SymbolicExecutionVisitor.java

```diff
@@ -60,20 +61,61 @@ public void visitNode(Tree tree) {
   @CheckForNull
   public MethodBehavior execute(MethodTree methodTree) {
     try {
-      MethodBehavior methodBehavior = behaviorCache.get(methodTree.symbol());
-      if(methodBehavior == null) {
-        methodBehavior = new MethodBehavior(methodTree.symbol());
-        behaviorCache.put(methodTree.symbol(), methodBehavior);
-        ExplodedGraphWalker walker = egwFactory.createWalker(this);
-        methodBehavior = walker.visitMethod(methodTree, methodBehavior);
-        methodBehavior.completed();
-      }
+      MethodBehavior methodBehavior = new MethodBehavior(methodTree.symbol());
+      behaviorCache.add(methodTree.symbol(), methodBehavior);
+      ExplodedGraphWalker walker = egwFactory.createWalker(behaviorCache);
+      methodBehavior = walker.visitMethod(methodTree, methodBehavior);
+      methodBehavior.completed();
       return methodBehavior;
     } catch (ExplodedGraphWalker.MaximumStepsReachedException | ExplodedGraphWalker.ExplodedGraphTooBigException | BinaryRelation.TransitiveRelationExceededException exception) {
       LOG.debug("Could not complete symbolic execution: ", exception);
     }
     return null;
   }
 
+  class BehaviorCache {
+    final Map<Symbol.MethodSymbol, MethodBehavior> behaviors = new LinkedHashMap<>();
+
+    void add(Symbol.MethodSymbol symbol, MethodBehavior behavior) {
+      behaviors.put(symbol, behavior);
+    }
+
+    public MethodBehavior get(Symbol.MethodSymbol symbol) {
+      if (!behaviors.containsKey(symbol)) {
+        if (isObjectsRequireNonNullMethod(symbol)) {
+          behaviors.put(symbol, createRequireNonNullBehavior(symbol));
+        } else {
+          MethodTree declaration = symbol.declaration();
+          if (declaration != null) {
+            SymbolicExecutionVisitor.this.execute(declaration);
+          }
+        }
+      }
+      return behaviors.get(symbol);
+    }
+
+    private boolean isObjectsRequireNonNullMethod(Symbol symbol) {
+      return symbol.owner().type().is("java.util.Objects") && "requireNonNull".equals(symbol.name());
+    }
+
+    private MethodBehavior createRequireNonNullBehavior(Symbol.MethodSymbol symbol) {
+      MethodBehavior behavior = new MethodBehavior(symbol);
+      MethodYield happyYield = new MethodYield(symbol.parameterTypes().size(), false);
+      happyYield.exception = false;
+      happyYield.parametersConstraints[0] = ObjectConstraint.notNull();
+      happyYield.resultIndex = 0;
+      happyYield.resultConstraint = happyYield.parametersConstraints[0];
+      behavior.addYield(happyYield);
+
+      MethodYield exceptionalYield = new MethodYield(symbol.parameterTypes().size(), false);
+      exceptionalYield.exception = true;
+      exceptionalYield.parametersConstraints[0] = ObjectConstraint.nullConstraint();
+      behavior.addYield(exceptionalYield);
```

Comment:
As discussed : for the time being we are not even sure we actually want to provide the exceptional yield for this case (to limit noise and/or FP).